docs/diagrams/replicator.ditaa

```diff
@@ -0,0 +1,63 @@
+                                                 <--------------------------------------------------                                         |      |
+                                                                                                                                             \------/
+
+
+
+                                                                   /----------\
+                                                                   |  UPLOAD  |
+                                                                   \----------/
+
+/------\
+|{s}   |   Find next file to upload
+|Shared+<-----------------------------------                        metadata
+|  +   |
+|Files +----------------------------------->        { _id, _rev, _revisions, attributes }                                                    /------\
+|      |                                         -------------------------------------------------->                                         |      |
+\------/                                                                                                 ----------------------------------->+Shared|
+                                                                                                                                             |   +  |
+                                                                                                         <-----------------------------------+ Files|
+                                                 <--------------------------------------------------                                         |      |
+                                                    { key∶ string }                                                                          \------/
+/------\
+|      |
+|      +<-----------------------------------
+|Swift |
+|      +----------------------------------->                          upload                                                                 /------\
+|      |                                         -------------------------------------------------->                                         |      |
+\------/                                                                                                 ----------------------------------->+      |
+                                                                                                                                             | Swift|
+                                                                                                         <-----------------------------------+      |
+                                                 <--------------------------------------------------                                         |      |
+                                                                                                                                             \------/
```

Comment:
On the generated png, the Shared and Shared+files blocks appear as databases on Alice, but not on Bob. Neither are foos, on both sides.

Reply:
On a side note, I find the png a bit difficult to read, the formatting is sometimes a bit weird/ugly